documentation/catalog/guides/adding_a_new_provider.md

```diff
@@ -128,12 +130,35 @@ PROVIDER_WORKFLOWS = [
 There are many other options that allow you to tweak the `schedule` (when and
 how often your DAG is run), timeouts for individual steps of the DAG, and more.
 These are documented in the definition of the `ProviderWorkflow` dataclass.
-_<TODO: add docs for other options.>_
+
+<!--TODO: add docs for other options.-->
 
 After adding your configuration, run `just up` and you should now have a fully
-functioning provider DAG! _<TODO: add and link to docs for how to run provider
-DAGs locally, preferably with images.>_
+functioning provider DAG!
+
+<!--TODO: add and link to docs for how to run provider DAGs locally, preferably with images.-->
 
 _NOTE_: when your code is merged, the DAG will become available in production
 but will be disabled by default. A contributor with Airflow access will need to
 manually turn the DAG on in production.
+
+## Testing guide
+
+### Steps
+
+1. Ensure you've gone through the [quickstart](/catalog/guides/quickstart.md).
+   Ensure that the Docker daemon is running.
+
+2. Run individual test by creating a testing session within Docker, then
+   selecting only the tests associated with the provider
```

Comment:
```suggestion
   selecting only the tests associated with the provider.
```

What is the difference between running a test in a testing session and running a test in Docker directly? It would be really nice to have a short explanation of why choose one over the other.

Reply:
@obulat opening a `test-session` in Docker drops you into a shell which is set up to run tests, so it's very easy to run tests over and over (while potentially modifying the code) without having to spin the Docker container up each time the tests need to be run. Running the tests on Docker directly (e.g. using `just catalog/test`) will spin up the container, run the selected tests (or all by default) and then stop and remove the container. That can be useful for ensuring that all tests pass if one does not need to iterate and check the test failures repeatedly.

@ngken0995 do you mind adding a small blurb after these instructions with the above notes to describe the difference between those two methods? Feel free to reword what I've said too 😄